monitoring/backup_collector.go

```diff
@@ -47,32 +47,38 @@ func (bc *backupCollector) Describe(ch chan<- *prometheus.Desc) {
 }
 
 func (bc *backupCollector) Collect(ch chan<- prometheus.Metric) {
-	backups, err := bc.ds.ListBackups()
+	defer func() {
+		if err := recover(); err != nil {
+			bc.logger.WithField("error", err).Warn("panic during collecting metrics")
+		}
+	}()
+
+	backups, err := bc.ds.ListBackupsRO()
 	if err != nil {
 		bc.logger.WithError(err).Warn("error during scrape")
 		return
 	}
 
-	for name, backup := range backups {
+	for _, backup := range backups {
 		ch <- prometheus.MustNewConstMetric(
 			bc.stateMetric.Desc,
 			bc.stateMetric.Type,
 			float64(convertBackupState(backup.Status.State)),
-			bc.currentNodeID,
-			name,
+			backup.Name,
+			backup.Status.VolumeName,
```

Comment:
The actual volume name is stored by label, you could refer to this [code](https://github.com/longhorn/longhorn-manager/blob/436bda78a955bdaf4bd9a73bc5fd4613f84bcb24/controller/backup_controller.go#L401-L407)